python/cugraph/link_analysis/pagerank_wrapper.pyx

```diff
@@ -65,22 +66,27 @@ def pagerank(input_graph, alpha=0.85, personalization=None, max_iter=100, tol=1.
         df['pagerank'][nstart['vertex']] = nstart['values']
         has_guess = <bool> 1
 
-    cdef gdf_column c_identifier_col = get_gdf_column_view(df['vertex'])
-    cdef gdf_column c_pagerank_col = get_gdf_column_view(df['pagerank'])
-    cdef gdf_column c_pers_vtx
-    cdef gdf_column c_pers_val
+    #TODO FIX ME when graph class is upgraded to remove gdf_column
+    cdef gdf_column c_identifier = get_gdf_column_view(df['vertex'])
 
-    g.transposedAdjList.get_vertex_identifiers(&c_identifier_col)
+    cdef uintptr_t c_pagerank_val = get_column_data_ptr(df['pagerank']._column)
+    cdef uintptr_t c_pers_vtx = <uintptr_t>NULL
+    cdef uintptr_t c_pers_val = <uintptr_t>NULL
+    cdef sz = 0
+
+    if personalization is not None:
+        sz = personalization['vertex'].shape[0]
+        c_pers_vtx = get_column_data_ptr(personalization['vertex']._column)
+        c_pers_val = get_column_data_ptr(personalization['values']._column)
     
-    if personalization is None:
-        c_pagerank.pagerank(g, &c_pagerank_col, <gdf_column*> NULL, <gdf_column*> NULL,
-                <float> alpha, <float> tol, <int> max_iter, has_guess)
-    else:
-        c_pers_vtx = get_gdf_column_view(personalization['vertex'])
-        c_pers_val = get_gdf_column_view(personalization['values'])
-        c_pagerank.pagerank(g, &c_pagerank_col, &c_pers_vtx, &c_pers_val,
-                <float> alpha, <float> tol, <int> max_iter, has_guess)
+    if (df['pagerank'].dtype is np.float32): 
```

Comment:
Right now it is going to the else loop since if (df['pagerank'].dtype is np.float32): is False and hence giving incorrect results with double. It should be changed to "if (df['pagerank'].dtype == np.float32):"

Reply:
Done